lldb/include/lldb/API/SBDebugger.h

```diff
@@ -106,44 +124,122 @@ class LLDB_API SBDebugger {
                                           bool &is_debugger_specific);
 #endif
 
+  /// Get structured progress data from an event.
+  ///
+  /// \param [in] event
+  ///   The event to extract the progress information from.
+  ///
+  /// \return
+  ///   A structured data object containing progress information, or an invalid
+  ///   SBStructuredData if the event was not a progress event.
   static lldb::SBStructuredData
   GetProgressDataFromEvent(const lldb::SBEvent &event);
 
+  /// Get diagnostic information from an event.
+  ///
+  /// \param [in] event
+  ///   The event to extract the diagnostic information from.
+  ///
+  /// \return
+  ///   A structured data object containing diagnostic information, or an
+  ///   invalid SBStructuredData if the event was not a diagnostic event.
   static lldb::SBStructuredData
   GetDiagnosticFromEvent(const lldb::SBEvent &event);
 
+  /// Assignment operator.
   lldb::SBDebugger &operator=(const lldb::SBDebugger &rhs);
 
+  /// Initialize LLDB and its subsystems.
+  ///
+  /// This function should be called before any other LLDB functions. It
+  /// initializes all required subsystems for proper LLDB functionality.
   static void Initialize();
```

Comment:
Should this be marked as deprecated since we should favor the overload with error handling ?

Reply:
We could, but I don't think anything in the initialization is fallible anymore now that we don't initialize a VFS there anymore. Which reminds me, `GetReproducerPath` definitely needs to be marked as depreciated, but that's orthogonal to this PR.